src/java.base/share/classes/java/lang/StackStreamFactory.java

```diff
@@ -657,68 +597,150 @@ public boolean tryAdvance(Consumer<? super StackFrame> action) {
-
-
-            // ------ subclass may override the following methods -------
-            /**
-             * Resizes the buffers for VM to fill in the next batch of stack frames.
-             * The next batch will start at the given startIndex with the maximum number
-             * of elements.
-             *
-             * <p> Subclass may override this method to manage the allocated buffers.
-             *
-             * @param startIndex the start index for the first frame of the next batch to fill in.
-             * @param elements the number of elements for the next batch to fill in.
-             *
-             */
-            @Override
-            void resize(int startIndex, int elements) {
-                if (!isActive())
-                    throw new IllegalStateException("inactive frame buffer can't be resized");
+        @Override
+        final boolean filter(int index) {
+            return stackFrames[index].declaringClass() == Continuation.class
+                    && "yield0".equals(stackFrames[index].getMethodName());
+        }
+    }
 
-                assert startIndex == START_POS :
-                       "bad start index " + startIndex + " expected " + START_POS;
+    static final class ClassFrameBuffer extends FrameBuffer<ClassFrameInfo> {
+        final StackWalker walker;
+        ClassFrameInfo[] classFrames;      // caller class for fast path
+        ClassFrameBuffer(StackWalker walker, int batchSize) {
+            super(batchSize);
+            this.walker = walker;
+            this.classFrames = new ClassFrameInfo[batchSize];
+            for (int i = START_POS; i < batchSize; i++) {
+                classFrames[i] = new ClassFrameInfo(walker);
+            }
+        }
 
-                int size = startIndex+elements;
-                if (classes.length < size) {
-                    // copy the elements in classes array to the newly allocated one.
-                    // classes[0] is a Thread object
-                    Class<?>[] prev = classes;
-                    classes = new Class<?>[size];
-                    System.arraycopy(prev, 0, classes, 0, startIndex);
-                }
-                currentBatchSize = size;
+        @Override
+        ClassFrameInfo[] frames() { return classFrames;}
+
+        @Override
+        ClassFrameInfo nextStackFrame() {
+            if (isEmpty()) {
+                throw new NoSuchElementException("origin=" + origin + " fence=" + fence);
+            }
+
+            ClassFrameInfo frame = classFrames[origin];
+            origin++;
+            return frame;
+        }
+
+        @Override
+        final Class<?> at(int index) { return classFrames[index].declaringClass();}
+
+        @Override
+        final boolean filter(int index) {
+            // no method information is available; should just filter
+            // "Continuation::yield0".
+            return classFrames[index].declaringClass() == Continuation.class;
```

Comment:
The Continuation frames around enter and yielding are hidden. Continuation.yield0 was the only frame that required special handling here, it may need to be re-checked to see if it is still required when using SHOW_HIDDEN_FRAMES.

Reply:
Looks like the filtering is not needed.   https://bugs.openjdk.org/browse/JDK-8304919 marked those methods with `@Hidden`.   If `SHOW_HIDDEN_FRAMES` is set, those frames should be included unless there is a reason not to.  

I will update the patch to remove the filtering in both `StackFrameBuffer` and `ClassFrameBuffer`.